src/wallet/wallet.cpp

```diff
@@ -3618,14 +3714,41 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
         walletInstance->SetMaxVersion(nMaxVersion);
     }
 
+
+    std::string hdWatchOnly = GetArg("-hdwatchonly", "");
+    CExtPubKey extPubKey;
+    if (!hdWatchOnly.empty()) {
+        CBitcoinExtPubKey bitcoinExtPubKey;
+        auto extpubKeyPrefix = chainparams.Base58Prefix(CChainParams::Base58Type::EXT_PUBLIC_KEY);
```

Comment:
Hmm.. I think this is incorrect. I could load a xpub (mainnet) on regtest.
Shouldn't `CBitcoinExtPubKey("<x|tpub>")` do the check for the correct encoding according to the chainparams?